### PR TITLE
fix(perf): Fix error messages coming from Gen. Query

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
The error messages don't always return in the same format, so this is actually an older bug. This will check whether the error is being returned as a string or an object {code, message, extra} with room to expand this in the future if we find more cases.

Refs JAVASCRIPT-26XX

### CODE REVIEW

**Review by udameli — APPROVED** [review]

**Review by edwardgou-sentry — APPROVED** [review]

**Review by dashed — APPROVED** [review]

File: tests/js/spec/utils/discover/discoverQuery.spec.jsx [hunk 1]
```diff
@@ -91,4 +91,70 @@ describe('DiscoverQuery', function () {
...
+      >
+        {({isLoading}) => {
+          if (isLoading) {
+            return 'loading';
+          }
+          return null;
+        }}
+      </DiscoverQuery>
+    );
+    await tick();
```
Comment (priscilawebdev) [inline tests/js/spec/utils/discover/discoverQuery.spec.jsx:121]: can we use `waitFor` or some async RTL query here?

File: tests/js/spec/utils/discover/discoverQuery.spec.jsx [hunk 2]
```diff
@@ -91,4 +91,70 @@ describe('DiscoverQuery', function () {
...
```
Comment (priscilawebdev) [inline tests/js/spec/utils/discover/discoverQuery.spec.jsx:156]: same here, so we don't need to use `await tick`

File: static/app/utils/discover/genericDiscoverQuery.tsx [hunk 3]
```diff
@@ -195,6 +205,32 @@ class _GenericDiscoverQuery<T, P> extends React.Component<Props<T, P>, State<T>>
     );
   };
 
+  /**
+   * The error type isn't consistent across APIs. We see detail as just string some times, other times as an object.
+   */
+  _parseError = (error: any): QueryError | null => {
```
Comment (priscilawebdev) [inline static/app/utils/discover/genericDiscoverQuery.tsx:211]: can we maybe reuse the function https://github.com/getsentry/sentry/blob/master/static/app/utils/handleXhrErrorResponse.tsx?

File: tests/js/spec/utils/discover/discoverQuery.spec.jsx (hunk 1 above)
Comment (k-fish) [reply tests/js/spec/utils/discover/discoverQuery.spec.jsx:121]: One day 😄  (I'll prob just convert the whole file at once in a future PR)

File: static/app/utils/discover/genericDiscoverQuery.tsx (hunk 3 above)
Comment (k-fish) [reply static/app/utils/discover/genericDiscoverQuery.tsx:211]: That's a good catch on that util function, I might give it a shot 👍

**Pull request merged**